src/boot/fwupdate.c

```diff
@@ -0,0 +1,586 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+/* Ported from:
+ * https://github.com/fwupd/fwupd-efi/blob/1a794fc7ce481176926687f4e88e6e0c8d5e9699/efi/fwupdate.c
+ * Copyright (C) 2014 Red Hat, Inc.
+ */
+
+#include "proto/graphics-output.h"
+#include "device-path-util.h"
+#include "efi.h"
+#include "efi-efivars.h"
+#include "efi-log.h"
+#include "fwupdate.h"
+#include "string-util-fundamental.h"
+#include "util.h"
+
+#define FWUPDATE_ATTEMPT_UPDATE 0x00000001
+#define FWUPDATE_ATTEMPTED      0x00000002
+
+#define CAPSULE_FLAGS_PERSIST_ACROSS_RESET    0x00010000
+#define CAPSULE_FLAGS_POPULATE_SYSTEM_TABLE   0x00020000
+#define CAPSULE_FLAGS_INITIATE_RESET          0x00040000
+
+/* The following types are defined by fwupd:
+ * https://github.com/fwupd/fwupd-efi/blob/1a794fc7ce481176926687f4e88e6e0c8d5e9699/efi/fwup-efi.h
+ */
+#define UX_CAPSULE_GUID {0x3b8c8162, 0x188c, 0x46a4, {0xae, 0xc9, 0xbe, 0x43, 0xf1, 0xd6, 0x56, 0x97} }
+
+typedef struct {
+        uint8_t version;
+        uint8_t checksum;
+        uint8_t image_type;
+        uint8_t reserved;
+        uint32_t mode;
+        uint32_t x_offset;
+        uint32_t y_offset;
+} _packed_ UX_CAPSULE_HEADER;
```

Comment:
oh, christ, it took me ages, but i found what this is, now:

https://learn.microsoft.com/en-us/windows-hardware/drivers/bringup/boot-screen-components

please fix the link above. 

googling for the definitions there, one finds this:

https://github.com/killvxk/nt-1/blob/299fe96583e030d850b34ae0b231eaffb630a84f/boot/osloader/efi/fwupdate.c#L62

any chance you can use the names of the fields, structures and defines from that?

Reply:
and please add a summarizing comment here explaining that this is for windows compatible graphical firmware updates. and that this is the extended header inserted right after the capsule header when the indicated GUID is used.

Reply:
> any chance you can use the names of the fields, structures and defines from that?

There is no license reference in that code, and actually it's (C) Microsoft, so it looks _really_ fishy? I'd really prefer not to copy anything from there, and instead use the same structure and naming as fwupd which has a clear copyright and a compatible license